lib/active_record_extended/query_methods/with_cte.rb

```diff
@@ -58,6 +72,8 @@ def pipe_cte_with!(value)
         def reset!
           @with_keys   = []
           @with_values = {}
+          @materialized_keys = []
+          @not_materialized_keys = []
```

Comment:
Seeing many of these arrays being used for simple key checking, we could convert these arrays into a `Set` (pretty sure it'd be a drop-in replacement without further code changes). We'd eliminate a lot of nested checking throughout the code base (i.e. `#materialized/1` & `#not_materialized/1`).